website/src/components/Edit/EditableSequences.ts

```diff
@@ -0,0 +1,127 @@
+import type { SequenceEntryToEdit } from '../../types/backend.ts';
+import type { ReferenceGenomesLightweightSchema } from '../../types/referencesGenomes.ts';
+
+type EditableSequenceFile = {
+    key: string;
+    label?: string;
+    value: string | null;
+    initialValue: string | null;
+};
+
+export class EditableSequences {
+    private static nextKey = 0;
+
+    private readonly editableSequenceFiles: EditableSequenceFile[];
+    private readonly maxNumberOfRows: number;
+
+    public get rows(): Required<EditableSequenceFile>[] {
+        const rows = this.editableSequenceFiles.map((row, i) => ({
+            ...row,
+            label: row.label ?? `Segment ${i + 1}`,
+        }));
+        if (rows.length < this.maxNumberOfRows) {
+            rows.push({
+                label: `Add a segment`,
+                value: null,
+                initialValue: null,
+                key: EditableSequences.getNextKey(),
+            });
+        }
+        return rows;
+    }
+
+    private constructor(rows: EditableSequenceFile[], maxNumberOfRows: number) {
+        this.editableSequenceFiles = rows;
+        this.maxNumberOfRows = maxNumberOfRows;
+    }
+
+    isMultiSegmented() {
+        return this.maxNumberOfRows > 1;
+    }
+
+    /**
+     * @param initialData The sequence entry to edit, from which the initial sequence data is taken.
+     * @param referenceGenomeLightweightSchema
+     */
+    static fromInitialData(
+        initialData: SequenceEntryToEdit,
+        referenceGenomeLightweightSchema: ReferenceGenomesLightweightSchema,
+    ): EditableSequences {
+        const existingDataRows = Object.entries(initialData.originalData.unalignedNucleotideSequences).map(
+            ([key, value]) => ({
+                label: `${key} segment`,
```

Comment:
After I click edit on a multi-segment sequence and then submit again. Preprocessing fails - from the logs it looks like the unalignedSequences is sent to preprocessing with the keys `M segment`, `L segment` etc. instead of `M` and `S`

<img width="2582" height="750" alt="image" src="https://github.com/user-attachments/assets/d522f24e-1348-4f24-bf28-6f772ef68f42" />

Reply:
That's because the code always showed "S segment" as the label there (when the segment name was "S"). But thinking about it, that probably doesn't make much sense anymore right? It would be better to show whatever the user used as FASTA header in the submission?